lib/Doctrine/ORM/Tools/Pagination/LimitSubqueryOutputWalker.php

```diff
@@ -100,6 +102,8 @@ public function __construct($query, $parserResult, array $queryComponents)
         $this->platform = $query->getEntityManager()->getConnection()->getDatabasePlatform();
         $this->rsm      = $parserResult->getResultSetMapping();
 
+        $query = clone $query;
```

Comment:
Avoids modifying the original query. If it has limit/offset reset, we cannot use it subsequently for the finalization phase.